networking/p2p/src/test/java/tech/pegasys/teku/networking/p2p/network/config/NetworkConfigTest.java

```diff
@@ -122,4 +129,37 @@ void checkDirectPeersConfigCreatedCorrectly() {
     assert manager.isDirectPeer(peerId1);
     assert !manager.isDirectPeer(peerId2);
   }
+
+  @Test
+  void failsIfInvalidNumberOfNetworkInterfacesAreSet() {
+    assertThatThrownBy(
+            () ->
+                NetworkConfig.builder()
+                    .networkInterfaces(List.of("0.0.0.0", "::", "1.2.3.4"))
+                    .build())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage("Invalid number of --p2p-interface. It should be either 1 or 2, but it was 3");
+  }
+
+  @Test
+  void failsIfTwoIPv4NetworkInterfacesAreSet() {
+    assertThatThrownBy(
+            () -> NetworkConfig.builder().networkInterfaces(List.of("0.0.0.0", "1.2.3.4")).build())
+        .isInstanceOf(InvalidConfigurationException.class)
+        .hasMessage(
+            "Expected an IPv4 and an IPv6 address for --p2p-interface but only [IP_V4] was set");
+  }
+
+  @Test
+  void failsIfTwoIPv6AdvertisedIpsAreSet() {
+    assertThatThrownBy(
+            () ->
+                NetworkConfig.builder()
+                    .advertisedIps(
+                        Optional.of(List.of("::", "2001:db8:3333:4444:5555:6666:7777:8888")))
+                    .build())
+        .isInstanceOf(InvalidConfigurationException.class)
+        .hasMessage(
+            "Expected an IPv4 and an IPv6 address for --p2p-advertised-ip but only [IP_V6] was set");
+  }
```

Comment:
is it tested the dual stack happy case too?

Reply:
Good point, will add a test for it tomorrow